climpred/stats.py

```diff
@@ -1,322 +1,14 @@
-                                  as well as p values.
-    Returns:
-        Pearson correlation coefficients
-        If return_p True, associated p values.
-
-    References:
-        * Wilks, Daniel S. Statistical methods in the atmospheric sciences.
-          Vol. 100. Academic press, 2011.
-        * Lovenduski, Nicole S., and Nicolas Gruber. "Impact of the Southern
-          Annular Mode on Southern Ocean circulation and biology." Geophysical
-          Research Letters 32.11 (2005).
-
-    """
-    if lag != 0:
-        N = x[dim].size
-        normal = x.isel({dim: slice(0, N - lag)})
-        shifted = y.isel({dim: slice(0 + lag, N)})
-        if dim not in list(x.coords):
-            normal[dim] = np.arange(1, N)
-        shifted[dim] = normal[dim]
-        r = pearson_r(normal, shifted, dim)
-    else:
-        r = pearson_r(x, y, dim)
-    if return_p:
-        p = _eff_p_value(x, y, r, dim)
-        # return with proper dimension labeling. would be easier with
-        # apply_ufunc, but having trouble getting it to work here. issue
-        # probably has to do with core dims.
-        dimlist = list(r.dims)
-        for i in range(len(dimlist)):
-            p = p.rename({'dim_' + str(i): dimlist[i]})
-        return r, p
-    else:
-        return r
-
-
-def _eff_p_value(x, y, r, dim):
-    """Computes p values accounting for autocorrelation in time series.
-
-    Args:
-        x (xarray object): Independent time series.
-        y (xarray object): Dependent time series.
-        r (xarray object): Pearson correlations between x and y.
-        dim (str): Dimension to compute compute p values over.
-
-    Returns:
-        p values accounting for autocorrelation in input time series.
-
-    References:
-        * Wilks, Daniel S. Statistical methods in the atmospheric sciences.
-          Vol. 100. Academic press, 2011.
-    """
-
-    def _compute_autocorr(v, dim, n):
-        """
-        Return normal and shifted time series
-        with equal dimensions so as not to
-        throw an error.
-        """
-        shifted = v.isel({dim: slice(1, n)})
-        normal = v.isel({dim: slice(0, n - 1)})
-        # see explanation in autocorr for this
-        if dim not in list(v.coords):
-            normal[dim] = np.arange(1, n)
-        shifted[dim] = normal[dim]
-        return pearson_r(shifted, normal, dim)
-
-    n = x[dim].size
-    # find autocorrelation
-    xa, ya = x - x.mean(dim), y - y.mean(dim)
-    xauto = _compute_autocorr(xa, dim, n)
-    yauto = _compute_autocorr(ya, dim, n)
-    # compute effective sample size
-    n_eff = n * (1 - xauto * yauto) / (1 + xauto * yauto)
-    n_eff = np.floor(n_eff)
-    # constrain n_eff to be at maximum the total number of samples
-    n_eff = n_eff.where(n_eff <= n, n)
-    # compute t-statistic
-    t = r * np.sqrt((n_eff - 2) / (1 - r ** 2))
-    p = xr.DataArray(ss.t.sf(np.abs(t), n_eff - 2) * 2)
-    return p
-
-
-@is_xarray(0)
-def rm_poly(ds, order, dim='time'):
```

Comment:
as mentioned on gitter I like this format and for rm_poly I dont see why `x,y,dim` would be needed